### ISSUE TITLE
CraftBeerPi Brewing Controller 4.0 (Version: 4.0.0.55)  Pi4B 8G CPU Load  test 

### ISSUE THREAD

**Issue by jsbjx** [avollkopf/craftbeerpi4#20]
![e84f05c82c6cecda1856ef27b4f325c](https://user-images.githubusercontent.com/32890201/145202709-23600cae-cfc7-49e7-ba8d-33c0383230e2.png)
Test with two versions: 
2021-05-07-raspios-buster-armhf
![H{({{L`3E5YG 03$_GBUFTD](https://user-images.githubusercontent.com/32890201/145202771-09d1716a-f904-4a9c-918b-cce63e39c547.png)
2021-05-07-raspios-buster-armhf-lite
![f4927fee3d7f62a5c844cca8f1e73d2](https://user-images.githubusercontent.com/32890201/145202799-6b729eff-d5ba-46db-b7b2-e598e5a9f3f9.png)
Desktop version As long as the water flow animation starts CPU Load 70% （yellow）
Server version water flow animation start has no Influence （blue）

**Comment by avollkopf** [comment]
I am not sure, if you are still talking about the charts topic in a dashboard or something else.
Charts in a dashboard are just not meant for production dashboards.

I am not sure how you defined the two different CPU load sensors, and what is the difference between server and desktop version in one chart?

What exactly are you referring to with 'water flow' -> path animations or something else?

**Comment by jsbjx** [comment]
I have deleted all the dashboards.
There are two types of Pi systems. One is that the HDMI port with a desktop can be directly connected to the monitor, and the other is without a desktop. Access web pages via PC.
What I want to explain is to install two versions on the same machine and start the animation. There is a significant difference in CPU load.

**Comment by avollkopf** [comment]
@jsbjx The cpu load just gives the overall CPU load information. Can you please also run 'top' from the command line and post the information here. This will give more insight into the individual processes that contribute to the overall CPU load.

**Comment by avollkopf** [comment]
I just added one 'pipe' to my dashboard for testing and do not see any increased CPU load. I can't reproduce it on my Pi4B. Majority of the load is consumed by chromium and not cbpi.

**Comment by jsbjx** [comment]
![fb3d0808cdaeec128adc5bd70bde669](https://user-images.githubusercontent.com/32890201/145838186-82617b77-eba0-4aa1-8348-ee0318a2444c.png)
![654ea1e712f5e018ca35a3f7ca064dc](https://user-images.githubusercontent.com/32890201/145838204-3f7cd802-c856-4089-b6f0-1185356e769a.png)

**Comment by jsbjx** [comment]
[cbpi4_config.zip](https://github.com/avollkopf/craftbeerpi4/files/7704843/cbpi4_config.zip)

**Comment by jsbjx** [comment]
![8281a11a6d384879d3b48db4e5b3f01](https://user-images.githubusercontent.com/32890201/145839789-fe3c58ac-f7a5-4051-b243-822dd3701f0b.png)
This is the water pipe is not activated

**Comment by jsbjx** [comment]
Chromium problem?

**Comment by kalausr** [comment]
Hi - I tried to see if the pipe animation is causing a higher load on my system. Here with my Dashboard:

![Screenshot 2021-12-13 at 22 25 19](https://user-images.githubusercontent.com/87930313/145891263-8a72d397-3a49-4761-bc64-980c1be5e2ec.png)
It did not affect the CPU Load significantly:
![D48183BC-5651-4392-9FE8-C4287C4D09F8](https://user-images.githubusercontent.com/87930313/145891400-b24c3852-3673-4579-b2d1-02251609beb1.jpeg)

Cheers, Klaus

**Comment by jsbjx** [comment]
If you log in and browse with a PC browser, you can see that the CPU Load will not change. But if you connect the Raspberry Pi HDMI to the monitor and open the browser to watch the water flow, the CPU Load will increase.

**Comment by avollkopf** [comment]
I can confirm now that CPU load is increasing when you use the browser on your pi. This is independent from chromium and can be also seen with firefox. I tried to change some settings in chromium to enable GPU support. However these settings are still experimental and enabling GPU support does even increase CPU usage. 

I tried to look into the path code at the UI side, but I can't change anything here as it is very complex and I did not write this code. It is also not related to the python based server but to the JS based UI in combination with a browser on the pi. 

A browser on a PC/Mac does also increase CPU usagen when running with path animations, but the increase is only 1-2 %. 

Hence, we need to close this issue on the server side as it is not related to the code. I also see no chance to fix things urrently on the ui side. Here we might either need JS experts or better GPU support fro chromium on the pi
